CMakeLists.txt

```diff
@@ -89,12 +89,11 @@ if (BUILD_PIPS_IPM OR BUILD_PIPS_NLP)
   find_package(OpenMP)
 endif()
 
-# include different "whole archive" linking options depending on compiler
-if (CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
-  set(WHOLE_ARCHIVE "-Wl,-all_load")
-  set(NO_WHOLE_ARCHIVE "-Wl,-noall_load")
-  message(STATUS "SETTING HAVE_CLANG")
-  set(HAVE_CLANG TRUE)
+# include different "whole archive" linking options depending on system
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  message(STATUS "Detected Darwin turning off --whole-archive")
+  set(WHOLE_ARCHIVE "")
```

Comment:
Although I could successfully compile `PIPS-NLP` on my mac, the previous setting `set(WHOLE_ARCHIVE "-Wl,-all_load")` also worked for me.

Reply:
Perhaps you have an older version of Xcode (that does the linking; Gnu only does the compiling) and it still supports this option? My linker said it didn't exist. (It's kind of a weird option anyways and probably in most circumstances doesn't matter either way).